src/pages/common/_model_loadsave.c

```diff
@@ -30,7 +30,7 @@ static int ini_handle_icon(void* user, const char* section, const char* name, co
 
 static int ini_handle_name(void* user, const char* section, const char* name, const char* value)
 {
-    long idx = (long)user;
+    int idx = (int)user;
     if(section[0] == '\0' && (strcasecmp(name, MODEL_NAME) == 0 || strcasecmp(name, MODEL_TEMPLATE) == 0)) {
```

Comment:
what is wrong with this?  sizeof(void*) == sizeof(long) on all platforms

Reply:
long should use %ld to printf. I am not sure if we want to do that in the following line. and idx to me int is large enough to hold idx.

Reply:
idx can be an int, but the cast should be to an equivalent sized pointer I think. intptr_t would work for the cast.(I didn't check, I assume we have both intptr_t and uintptr_t)